src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs

```diff
@@ -478,7 +478,30 @@ public static IEnumerator GetEnumerator(IsolatedStorageScope scope)
             // (say, for a particular StrongName). You could iterate any store opened at least once by NetFX on
             // NetFX as it would create the needed identity file. You wouldn't be able to iterate if it was only
             // ever opened by CoreFX, as the needed file isn't there yet.
-            throw new PlatformNotSupportedException();
+            return new IsolatedStorageFileEnumerator();
```

Comment:
Instead of adding a no-op `IsolatedStorageFileEnumerator` class, could this be:

```c#
public static IEnumerator GetEnumerator(IsolatedStorageScope scope) =>
    Array.Empty<IsolatedStorageFile>().GetEnumerator();
```

?